### PR TITLE
chore: Change debug type to "embedded" instead of "PDB"

### PR DESCRIPTION
**Pull request by tippmar-nr**
This change is to enabled embedded PDBs in our debug builds of ILRepacked assemblies, to facilitate local debugging. 

This is intended to be a temporary change, pending resolution of an [issue where the DLL is written with an incorrect path to the .PDB file](https://github.com/gluck/il-repack/issues/352). 

There is no change to the release build at all.

### CODE REVIEW

**Review by chynesNR — APPROVED** [review]

**Review by nr-ahemsath — APPROVED** [review]
Good job figuring this out and filing the issue in their repo. 💯

**Comment by tippmar-nr** [comment]
> Good job figuring this out and filing the issue in their repo. 💯

@nr-ahemsath it was actually @chynesNR who figured out the pathing problem. I just did the paperwork... ;-)

File: tests/Agent/IntegrationTests/SharedApplications/Common/SharedApplicationHelpers/ReflectionHelpers/ReflectionUtil.cs [hunk 1]
```diff
@@ -79,7 +79,18 @@ public static List<Type> GetClassesByType<T>(Assembly assembly)
...
-                foreach (var type in assembly.GetTypes())
+                Type[] types;
+                try
+                {
+                    types = assembly.GetTypes();
+                }
+                catch (Exception)
+                {
+                    Console.WriteLine($"Exception calling .GetTypes on assembly {assembly.FullName}");
+                    throw;
```
Comment (chynesNR) [inline tests/Agent/IntegrationTests/SharedApplications/Common/SharedApplicationHelpers/ReflectionHelpers/ReflectionUtil.cs:90]: Don't we want to continue if we hit a bad assembly? Or do you want it to fail out?

**Comment by nrcventura** [comment]
Are we sure that we want to do this? We found a workaround of manually adding the build folders as a symbol location. We don't know when or if a fix will be delivered. I think that I'd rather keep our dependencies more up-to-date.

File: tests/Agent/IntegrationTests/SharedApplications/Common/SharedApplicationHelpers/ReflectionHelpers/ReflectionUtil.cs (hunk 1 above)
Comment (tippmar-nr) [reply tests/Agent/IntegrationTests/SharedApplications/Common/SharedApplicationHelpers/ReflectionHelpers/ReflectionUtil.cs:90]: I'd rather have it die, as it's a sign of a bigger problem.

**Comment by tippmar-nr** [comment]
> Are we sure that we want to do this? We found a workaround of manually adding the build folders as a symbol location. We don't know when or if a fix will be delivered. I think that I'd rather keep our dependencies more up-to-date.

@nrcventura I think I've found a better way to deal with the issue - the author of ILRepack [acknowledged the bug and suggests a workaround](https://github.com/gluck/il-repack/issues/352#issuecomment-1945267824) of setting `<DebugType>` to `embedded` for our ILRepack'ed assemblies, which will avoid the issue with finding PDBs entirely. 

I'm testing that now but it looks promising.

**Review by nrcventura — APPROVED** [review]

**Review by chynesNR — APPROVED** [review]

**Comment by KirillOsenkov** [comment]
I recommend that you always use embedded pdbs, even in Release. There are numerous benefits and no known downsides. You'll get line numbers in exception stack traces in production, and never have to worry about locating the pdb files (which might be in a different location).

The problem with https://github.com/gluck/il-repack/issues/352 is that even if ILRepack fixes it somehow, the path to the pdb embedded in the dll is just a hint to the debugger anyway. It won't work if you're on a different machine, or if you otherwise move the dll and the pdb from the original location.

**Comment by KirillOsenkov** [comment]
Can you catch me up on what the actual problem was? The VS debugger should be able to pick up the .pdb if it's next to the .dll. Were you not copying the .pdb next to the .dll? ILRepack copies both the .dll and the .pdb from the temporary folder that gets deleted into the output directory.

**Comment by chynesNR** [comment]
Hi @KirillOsenkov  -- thanks for checking in. The issue was with local debugging. We build our binaries, then do an ILRepack, then have a copy step that places the DLLs in a central location (without the PDBs). When attaching Visual Studio to a process using those libraries, the hint path was wrong, pointing to a directory that didn't exist, and we had to manually locate the PDBs in the ILRepack folder. Certainly not the end of the world, but it cost us some time that day as we tried to figure out what had changed, and discovered it was ILRepack. And since the hint path was pointing to a directory that didn't exist, and everything was run locally, it felt like a bug.

**Comment by KirillOsenkov** [comment]
Makes sense! Yes, just embed pdbs always and this class of issues will go away.

**Pull request merged**